curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5efc4f528d6a74d05e68af74.md

```diff
@@ -9,73 +9,90 @@ dashedName: step-56
 
 There's another way to associate an `input` element's text with the element itself. You can nest the text within a `label` element and add a `for` attribute with the same value as the `input` element's `id` attribute.
 
-Here is an example of using the `for` attribute to associate a `label` with an `input` element:
+Given an `input` element as below:
 
 ```html
-<label for="breakfast"> Breakfast </label>
 <input id="breakfast" type="radio" name="meal" value="breakfast">
 ```
 
+An example of a `label` element that is associated to this `input` element is:
+
+```html
+<label for="breakfast">Breakfast</label>
+```
+
 Associate the text `Loving` with the checkbox by nesting only the text `Loving` in a `label` element and giving it an appropriate `for` attribute.
 
 # --hints--
 
-You should make sure the checkbox is still present.
+You should make sure the checkbox element is still present, and you did not make any changes to it. It should stay as it was in the starting code.
 
 ```js
-assert($('input[type="checkbox"]')[0]);
+const checkboxElementRegex = /\s*<input\s+id\s*=\s*"loving"\s*type\s*=\s*"checkbox"\s*>/;
```

Comment:
Do we need to revert to a regex?

I understand the goal is to keep campers from editing the input element, but if they add spacing do we really need to care? As long as the attributes and DOM layout are what we want?

Reply:
The problem was that some changes to the input element, caused us to assert the wrong thing.

```
<input id="loving" type="checkbox" <label>Loving</label>
```
> The text Loving should be wrapped in a label element.

With the regex it is this instead:

> You should make sure the checkbox element is still present, and you did not make any changes to it. It should stay as it was in the starting code.

Reply:
Gosh darn browsers and their ability to parse malformed code 🤣